src/pairinteraction_gui/worker.py

```diff
@@ -136,16 +168,21 @@ def run(self) -> None:
         return result
 
     @classmethod
-    def terminate_all(cls) -> None:
+    def terminate_all(cls, *, create_new_pool_async: bool = False) -> None:
         """Terminate all processes."""
+        if cls._async_worker is not None:
+            cls._async_worker.join()
+
         if cls._pool is None:
             return
 
         cls._pool.terminate()
         cls._pool = None
-
         logger.debug("Process pool terminated.")
 
+        if create_new_pool_async:
```

Comment:
create_new_pool_async -> create_new_pool. The relevant aspect of this parameter is whether a new pool should be created or not, and not whether it is async or not.